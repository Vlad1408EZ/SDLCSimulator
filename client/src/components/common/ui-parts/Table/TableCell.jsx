import React from "react";
import clx from "classnames";
import s from "./table.module.scss";

const TableCell = ({ value, isFirstCell = false }) => {
	const newVal =
		typeof value === "number" ? value.toFixed(2) : !value ? "-/-" : value;
	return (
		<div className={clx(s.tableCell, isFirstCell && s.first)}>
			<span>{newVal}</span>
		</div>
	);
};

export default TableCell;
