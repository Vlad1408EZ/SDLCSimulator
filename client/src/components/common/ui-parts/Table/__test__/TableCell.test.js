import React from 'react';
import renderer from 'react-test-renderer';

import TableCell from "../TableCell"

it("Table cell with number has 2 values after 0", () => {
    const tree = renderer
        .create(<TableCell value={123.1454} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
})

it("Table cell with empty string equal to \"-/-\"", () => {
    const tree = renderer
        .create(<TableCell value="" />)
        .toJSON();
    expect(tree).toMatchSnapshot();
})