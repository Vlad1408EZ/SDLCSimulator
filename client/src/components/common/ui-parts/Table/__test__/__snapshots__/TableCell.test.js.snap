// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table cell with empty string equal to "-/-" 1`] = `
<div
  className="tableCell"
>
  <span>
    -/-
  </span>
</div>
`;

exports[`Table cell with number has 2 values after 0 1`] = `
<div
  className="tableCell"
>
  <span>
    123.15
  </span>
</div>
`;
